plugins/ipam/host-local/backend/allocator/pool.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package allocator
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+func (p *Pool) Contains(addr net.IP) error {
+	if err := canonicalizeIP(&addr); err != nil {
+		return err
+	}
+
+	for _, r := range *p {
+		if r.Contains(addr) == nil {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("%s not in range set", addr.String())
+}
+
+// RangeFor finds the range that contains an IP, or nil if not found
+func (p *Pool) RangeFor(addr net.IP) *Range {
```

Comment:
Should we canonicalize the IP here too?  If so, could just implement Contains() on top of RangeFor()